packages/lodestar-utils/src/bytes.ts

```diff
@@ -1,19 +1,96 @@
 import {toBufferLE, toBigIntLE, toBufferBE, toBigIntBE} from "bigint-buffer";
+import {copyFromBuf64LE, copyFromBuf64BE, copyToBuf64LE, copyToBuf64BE} from "./buf64copy";
 
 type Endianness = "le" | "be";
 
 /**
  * Return a byte array from a number or BigInt
  */
 export function intToBytes(value: bigint | number, length: number, endianness: Endianness = "le"): Buffer {
-  return bigIntToBytes(BigInt(value), length, endianness);
+  switch (typeof value) {
+    case "number":
+      return numberToBytes(value, length, endianness);
+    case "bigint":
+      return bigIntToBytes(value, length, endianness);
+      break;
+    default:
+      throw new Error(`unsupported number type ${typeof value}`);
+  }
 }
 
 /**
  * Convert byte array in LE to integer.
  */
+
+const bufferG = new ArrayBuffer(8);
+const viewG = new DataView(bufferG);
+
+export function numberToBytes(value: number, length: number, endianness: Endianness = "le"): Buffer {
+  if (length > 8 || value > Number.MAX_SAFE_INTEGER || value < -Number.MAX_SAFE_INTEGER)
+    return intToBytes(BigInt(value), length, endianness);
+  //can't handle if above conds are true
+  //length less than required will lead to truncation on the msb side, same as bigint inttobytes
+
+  if (value < 0) value = -value; //bigint inttobytes always return the absolute value
```

Comment:
this should be rethought. perhaps our top-level functions should operate on "uint"s instead of ints, this would match the spec closer, too.